util/llbutil/authprovider/cloudauth/ecr_login.go

```diff
@@ -0,0 +1,70 @@
+package cloudauth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login/api"
+	"github.com/docker/cli/cli/config/types"
+)
+
+func (ap *authProvider) getAuthConfigECR(ctx context.Context, fullPathPrefix, pathPrefix, org, project, host string) (*authConfig, error) {
+	registry, err := api.ExtractRegistry(host)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract aws ecr registry details from %s: %w", host, err)
+	}
+
+	if registry.FIPS {
+		return nil, fmt.Errorf("FIPS registry %s not supported", host)
+	}
+
+	accessKeyIDPath := pathPrefix + "AWS_ACCESS_KEY_ID"
+	secretAccessKeyPath := pathPrefix + "AWS_SECRET_ACCESS_KEY"
+
+	ap.console.VerbosePrintf("looking up %sAWS_ACCESS_KEY_ID", fullPathPrefix)
+	accessKeyID, err := ap.getProjectOrUserSecret(ctx, org, project, accessKeyIDPath)
+	if err != nil {
+		return nil, err
+	}
+	accessKeyID = strings.TrimSpace(accessKeyID)
+	if accessKeyID == "" {
+		return nil, fmt.Errorf("%sAWS_ACCESS_KEY_ID is missing (or empty), but %scred_helper was set to ecr-login", fullPathPrefix, fullPathPrefix)
```

Comment:
`%[1]s` can be used to reuse the first argument to a formatting function. But we may also want to use `path.Join` here.

Reply:
> %[1]s

neat! I'll keep that in mind for next time.

I actually rewrote a lot of this path joining code to make it easier to follow